### PR TITLE
[SITE] Document cache.expiration-interval-ms in Spark Configuration

### PR DESCRIPTION
**Pull request by kbendick**
Adds documentation to the site for the catalog config key `cache.expiration-interval-ms` for Spark.

This has not been implemented for Flink yet, so I've not added it to the Flink documentation (working on that in a separate PR).

Here's a reference to the JavaDoc for this configuration value: https://github.com/apache/iceberg/blob/9c3e340cd67699c3d2499762d794ab1bc1ee7f45/core/src/main/java/org/apache/iceberg/CatalogProperties.java#L42-L52

### CODE REVIEW

**Comment by kbendick** [comment]
cc @rdblue @nastra @samredai @jackye1995 @RussellSpitzer added some documentation for `cache.expiration-interval-ms`.

I still need to plumb it through Flink, which I'm working on today and will update the Flink docs afterwards.

**Comment by kbendick** [comment]
image

File: site/docs/spark-configuration.md [hunk 1]
```diff
@@ -52,14 +52,15 @@ Iceberg supplies two implementations:
...
-| spark.sql.catalog._catalog-name_.warehouse         | hdfs://nn:8020/warehouse/path | Base path for the warehouse directory |
-| spark.sql.catalog._catalog-name_.cache-enabled     | `true` or `false`             | Whether to enable catalog cache, default value is `true` |
+| Property                                                      | Values                        | Description                                                                                                                                                                                                                                                                                                                                  |
+|---------------------------------------------------------------|-------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (jackye1995) [inline site/docs/spark-configuration.md:56]: there is no need to change all the other lines to have the same size description blocks, the website will format them correctly anyway. Could you remove the unrelated changes?

Comment (kbendick) [reply site/docs/spark-configuration.md:56]: Sure.

Comment (kbendick) [reply site/docs/spark-configuration.md]: Updated to just one line change.

File: site/docs/spark-configuration.md [hunk 2]
```diff
@@ -60,6 +60,7 @@ Both catalogs are configured using properties nested under the catalog name. Com
 | spark.sql.catalog._catalog-name_.uri               | thrift://host:port            | Metastore connect URI; default from `hive-site.xml` |
 | spark.sql.catalog._catalog-name_.warehouse         | hdfs://nn:8020/warehouse/path | Base path for the warehouse directory |
 | spark.sql.catalog._catalog-name_.cache-enabled     | `true` or `false`             | Whether to enable catalog cache, default value is `true` |
+| spark.sql.catalog._catalog-name_.cache.expiration-interval-ms | `30000` (30 seconds)          | Controls the duration for which entries in the catalog are cached; positive values will enable entries to be expired from the catalog cache after that interval if not accessed,`-1` disables cache expiration (same behavior as Iceberg prior to version 0.13.0), and `0` turns off the catalog cache entirely. Default is `30000` (30 sec) ||                                                    |                               |                                                          |
```
Comment (jackye1995) [inline site/docs/spark-configuration.md:63]: nit: space before `-1`.

Also I think the config name itself is already pretty self-explanatory, we don't need multiple lines explaining that is cache expiration. We can just point out the important parts which are (1) it's only effective when `cache-enabled` is `true`, and interval equals `0` essentially disables caching.

Comment (kbendick) [reply site/docs/spark-configuration.md]: Ok I've updated it to be shorter. Let me know what you think.

**Review by rdblue — APPROVED** [review]

File: site/docs/spark-configuration.md (hunk 1 above)
Comment (rdblue) [reply site/docs/spark-configuration.md]: Looks good! Thanks for updating this.

**Comment by rdblue** [comment]
Thanks, @kbendick!

**Review by nastra — COMMENTED** [review]
belated +1
